cpu/stm32f3/periph/pwm.c

```diff
@@ -134,37 +142,44 @@ int pwm_set(pwm_t dev, int channel, unsigned int value)
 
     switch (dev) {
 #if PWM_0_EN
+
         case PWM_0:
             tim = PWM_0_DEV;
             break;
 #endif
 #if PWM_1_EN
+
         case PWM_1:
             tim = PWM_1_DEV;
             break;
 #endif
+
         default:
             return -1;
     }
 
-    /* norm value to maximum possible value */
-    if (value > 0xffff) {
-        value = 0xffff;
+    /* norm value to actual resolution */
+    if (value > tim->ARR) {
+        value = tim->ARR;
```

Comment:
If you initialized your PWM with a resolution of e.g. 1000 in `pwm_init()` then `tim->ARR` is set to 999, not 0xffff. And this is the value your duty cycle must not exceed.